enterprise_subsidy/apps/subsidy/models.py

```diff
@@ -296,64 +323,99 @@ class AccessMethods:
     access_method = AccessMethods.DIRECT
 
     @property
     def group_client(self):
+        """
+        Return an API client for the groups service.
+        """
         return MOCK_GROUP_CLIENT
 
     @classmethod
     def get_policies_for_groups(cls, group_uuids):
+        """
+        Fetch all the policies that are related to the given groups.
+        """
         return cls.objects.filter(group_uuid__in=group_uuids)
 
     @property
     def catalog_client(self):
+        """
+        Return an API client for the enterprise-catalog service.
+        """
         return MOCK_CATALOG_CLIENT
 
     def is_entitled(self, learner_id, content_key):
+        """
+        Check that a given learner is entitled to enroll in the given content.
+        """
         if not self.catalog_client.catalog_contains_content(self.catalog_uuid, content_key):
             return False
-
         if not self.group_client.group_contains_learner(self.group_uuid, learner_id):
             return False
+        if not self.subsidy.is_redeemable(content_key):  # pylint: disable=no-member
+            return False
+        return True
 
-        elif self.subsidy.is_redeemable(learner_id, content_key):
-            return True
-
-        return False
+    def redeem_entitlement(self, learner_id, content_key):
+        """
+        Redeem an entitlement for the given learner and content.
 
-    def use_entitlement(self, learner_id, content_key):
+        Returns:
+            A ledger transaction, or None if the entitlement was not redeemed.
+        """
         if self.is_entitled(learner_id, content_key):
             if self.access_method == AccessMethods.DIRECT:
                 return self.subsidy.redeem(learner_id, content_key)
             if self.access_method == AccessMethods.REQUEST:
-                self.subsidy.request_redemption(learner_id, content_key)
-        return False
+                return self.subsidy.request_redemption(learner_id, content_key)
+        return None
 
-    def has_used_entitlement(self, learner_id, content_key):
+    def has_redeemed_entitlement(self, learner_id, content_key):
         if self.access_method == AccessMethods.DIRECT:
-            return self.subsidy.has_redeemed(learner_id, content_key)
-        if self.access_method == AccessMethods.REQUEST:
+            return self.subsidy.has_redeemed(learner_id, content_key)  # pylint: disable=no-member
+        elif self.access_method == AccessMethods.REQUEST:
             return self.subsidy.has_requested(learner_id, content_key)
+        else:
+            raise ValueError(f"unknown access method {self.access_method}")
 
 
 class SubscriptionAccessPolicy(SubsidyAccessPolicy):
     """
+    A subsidy access policy for subscriptions (all you can eat).
+
+    .. no_pii:
     """
     subsidy = models.ForeignKey(SubscriptionSubsidy, null=True, on_delete=models.SET_NULL)
 
 
 class LearnerCreditAccessPolicy(SubsidyAccessPolicy):
```

Comment:
this code was part of the example code but it is meant for the subsidy access policy API, we can remove this.